modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -821,6 +845,322 @@ MUL_SIMD(float, float)
+                                        v_round(a2 + s2),
+                                        v_round(a3 + s3),
+                                        v_round(a4 + s1),
+                                        v_round(a5 + s2),
+                                        v_round(a6 + s3));
+        }
+
+        if (x < length)
+        {
+            x = length - lanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename SRC>
+CV_ALWAYS_INLINE int addc_simd_c3_impl(const SRC in[], uchar out[],
+                                       const v_float32& s1, const v_float32& s2,
+                                       const v_float32& s3, const int length)
+{
+    constexpr int nlanes = v_uint8::nlanes;
+    constexpr int chan = 3;
+    constexpr int lanes = chan * nlanes;
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - lanes; x += lanes)
+        {
+            vx_store(&out[x],
+                     v_pack_u(v_pack(v_round(vg_load_f32(&in[x]) + s1),
+                                     v_round(vg_load_f32(&in[x + nlanes/4]) + s2)),
+                              v_pack(v_round(vg_load_f32(&in[x + nlanes/2]) + s3),
+                                     v_round(vg_load_f32(&in[x + 3*nlanes/4]) + s1))));
+
+            vx_store(&out[x + nlanes],
+                     v_pack_u(v_pack(v_round(vg_load_f32(&in[x + nlanes]) + s2),
+                                     v_round(vg_load_f32(&in[x + 5*nlanes/4]) + s3)),
+                              v_pack(v_round(vg_load_f32(&in[x + 3*nlanes/2]) + s1),
+                                     v_round(vg_load_f32(&in[x + 7*nlanes/4]) + s2))));
+
+            vx_store(&out[x + 2 * nlanes],
+                     v_pack_u(v_pack(v_round(vg_load_f32(&in[x + 2*nlanes]) + s3),
+                                     v_round(vg_load_f32(&in[x + 9*nlanes/4]) + s1)),
+                              v_pack(v_round(vg_load_f32(&in[x + 5*nlanes/2]) + s2),
+                                     v_round(vg_load_f32(&in[x + 11*nlanes/4]) + s3))));
+        }
+
+        if (x < length)
+        {
+            x = length - lanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename SRC>
+CV_ALWAYS_INLINE int addc_simd_c3_impl(const SRC in[], float out[],
+                                       const v_float32& s1, const v_float32& s2,
+                                       const v_float32& s3, const int length)
+{
+    constexpr int nlanes = v_float32::nlanes;
+    constexpr int chan = 3;
+    constexpr int lanes = chan * nlanes;
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - lanes; x += lanes)
```

Comment:
looks like this copy-paste part with lanes calculation & internal for-loop could be incorporated in  initial  dispatcher `addc_simd_c3(const SRC in[], const float scalar[], DST out[], const int length)` then
this `*_impl` would just preform intrisict calculation only without this dupliacted pre- and post-  operations

Reply:
Applied.